src/api/supabase/queries/user_queries.ts

```diff
@@ -16,87 +16,166 @@ const supabaseApiKey = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;
 // Initialize the Supabase client
 const supabase = createClient(supabaseUrl ?? '', supabaseApiKey ?? '');
 
-export async function fetchUserData(): Promise<
-  PostgrestSingleResponse<User[]> | { data: never[]; error: PostgrestError }
-> {
-  try {
-    const { data: users, error } = await supabase
-      .from('profiles')
-      .select('*')
-      .single();
-
-    if (error) {
-      console.error('Error fetching data:', error);
-      return { data: [], error };
-    }
+export async function fetchUserData() {
+  const { data, error }: { data: User[] | null; error: PostgrestError | null } =
+    await supabase.from('profiles').select('*');
 
-    return { data: users } as PostgrestSingleResponse<User[]>;
-  } catch (error) {
-    console.error('Error:', error);
-    throw error;
+  if (error) {
+    throw new Error(`An error occured when trying to read profiles: ${error}`);
```

Comment:
[nit] standardize the errors for all functions. i think some are using ${error} and some are using ${error.message}. i think either is fine honestly